js/ui/html_editor/modules/imageCursor.js

```diff
@@ -0,0 +1,58 @@
+import Quill from 'devextreme-quill';
+import BaseModule from './base';
+import eventsEngine from '../../../events/core/events_engine';
+import { addNamespace } from '../../../events/utils/index';
+
+const MODULE_NAMESPACE = 'dxHtmlEditorImageCursor';
+const clickEvent = addNamespace('dxclick', MODULE_NAMESPACE);
+
+let ImageCursorModule = BaseModule;
+
+if(Quill) {
+    ImageCursorModule = class ImageCursorModule extends BaseModule {
+        constructor(quill, options) {
+            super(quill, options);
+
+            this.addCleanCallback(this.clean.bind(this));
+            this._attachEvents();
+        }
+
+        _attachEvents() {
+            eventsEngine.on(this.quill.root, clickEvent, this._clickHandler.bind(this));
+        }
+
+        _detachEvents() {
+            eventsEngine.off(this.quill.root, clickEvent);
+        }
+
+        _clickHandler(e) {
+            if(this._isAllowedTarget(e.target)) {
+                this._adjustSelection(e);
+            }
+        }
+
+        _isAllowedTarget(targetElement) {
+            return this._isImage(targetElement);
+        }
+
+        _isImage(targetElement) {
+            return targetElement.tagName.toUpperCase() === 'IMG';
```

Comment:
Isn't `tagName` property already always in upper case? https://developer.mozilla.org/en-US/docs/Web/API/Element/tagName

Reply:
Or it's for XHTML, all right then, sorry :)